test/test_terraform.py

```diff
@@ -123,13 +137,30 @@ def test_generate_cmd_string(self, method, expected):
             assert s in result
 
     @pytest.mark.parametrize(*CMD_CASES)
-    def test_cmd(self, method, expected_output, expected_ret_code, expected_logs, string_logger, folder):
+    def test_cmd(self, method, expected_output, expected_ret_code, expected_exception, expected_logs, string_logger, folder):
         tf = Terraform(working_dir=current_path)
         tf.init(folder)
-        ret, out, err = method(tf)
+        try:
+          ret, out, err = method(tf)
+          assert not expected_exception
+        except TerraformCommandError as e:
+          assert expected_exception
+          ret = e.returncode
+          out = e.out
+          err = e.err
+          
         logs = string_logger()
         logs = logs.replace('\n', '')
-        assert expected_output in out
+        if isinstance(expected_output, list):
+            ok = False
+            for xo in expected_output:
+                if xo in out:
+                    ok = True
+                    break
+            if not ok:
+              assert expected_output[0] in out
```

Comment:
NIT: 4 instead of 2 sapce

Reply:
Good catch, my editor is set for 2 and I didn't see all of them